### PR TITLE
obs: gstreamer plugin

### PR DESCRIPTION
**Pull request by yurkobb**
###### Motivation for this change
Add the [obs-gstreamer](https://github.com/fzwoch/obs-gstreamer) OBS plugin (useful for things like attaching sources OBS doesn't support natively, but gstreamer does, like firewire (H)DV cameras, among other things).

###### Need help with
Currently there are two issues with path lookup:
1. [obs doesn't know about plugins outside its own package](https://github.com/NixOS/nixpkgs/issues/77627) - can be worked around by `ln -s ~/.nix-profile/lib/obs-plugins/obs-gstreamer.so ~/.config/obs-studio/plugin/obs-gstreamer/bin/64bit/`
2. obs doesn't know where to look for gstreamer plugins, required by this package. Can be worked around by looking up the store path of gstreamer and gstreamer-plugins and doing
```
export GST_PLUGIN_SYSTEM_PATH_1_0=/nix/store/789ac60d8hdyhgbi5arx26m34zp410ik-gst-plugins-base-1.16.2/lib/:/nix/store/2j71zkvld1r8vhyk44xilcid9sybjpqw-gstreamer-1.16.2/lib
```

I am not sure if #1 needs patching OBS to support loading plugins from the system and user profiles (at least i can't find any environment variables to tweak [by searching for getenv calls](https://github.com/obsproject/obs-studio/search?q=getenv&unscoped_q=getenv) in its source code).

About #2 - have no idea how to fix this... Still a nixpkgs newbie...

(cc @jb55 @MP2E from `default.nix` maintainers).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yurkobb** [comment]
Hm, since #2 is actually similar with other system/user-wide plugins [like LV2](https://github.com/musnix/musnix/blob/master/modules/base.nix#L68), and #1 [applies to any obs plugin](https://github.com/NixOS/nixpkgs/issues/77627), not just this one, I think I'm going to submit this as is.

File: pkgs/applications/video/obs-studio/obs-gstreamer.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    owner = "fzwoch";
+    repo = "obs-gstreamer";
+    rev = version;
+    sha256 = "1szfx5p2lb953blzw7prfd0nngjlwv2wn7jmnwvlssc9ci6jl6s8";
+  };
+
+  nativeBuildInputs = [ meson ninja pkgconfig ];
+  buildInputs = [ obs-studio gstreamer gst-plugins-base ];
+
+  meta = with lib; {
```
Comment (Lassulus) [inline pkgs/applications/video/obs-studio/obs-gstreamer.nix:26]: please add a maintainer

File: pkgs/applications/video/obs-studio/obs-gstreamer.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+, gst-plugins-base
+, pkgconfig
+, meson
+, ninja
+, obs-studio
+}:
+
+stdenv.mkDerivation rec {
+  pname = "obs-gstreamer";
+  version = "v0.1.0";
```
Comment (jb55) [inline pkgs/applications/video/obs-studio/obs-gstreamer.nix:14]: no v in version :wink: 
```suggestion
  version = "0.1.0";
```

**Review by jb55 — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/obs-studio/obs-gstreamer.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchFromGitHub {
+    owner = "fzwoch";
+    repo = "obs-gstreamer";
+    rev = version;
```
Comment (jb55) [inline pkgs/applications/video/obs-studio/obs-gstreamer.nix:19]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/video/obs-studio/obs-gstreamer.nix (hunk 2 above)
Comment (yurkobb) [reply pkgs/applications/video/obs-studio/obs-gstreamer.nix]: thank you, committed.

File: pkgs/applications/video/obs-studio/obs-gstreamer.nix (hunk 1 above)
Comment (yurkobb) [reply pkgs/applications/video/obs-studio/obs-gstreamer.nix:26]: I guess I should add myself then

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -8919,6 +8919,16 @@
     githubId = 687198;
     name = "Yuri Aisaka";
   };
+  yurkobb = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8922]: Please put this into a separate commit with the message ``maintainers: add yurkobb``.

File: pkgs/applications/video/obs-studio/obs-gstreamer.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "fzwoch";
+    repo = "obs-gstreamer";
+    rev = "v${version}";
+    sha256 = "1szfx5p2lb953blzw7prfd0nngjlwv2wn7jmnwvlssc9ci6jl6s8";
+  };
+
+  nativeBuildInputs = [ meson ninja pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/obs-gstreamer.nix:23]:
```suggestion
  nativeBuildInputs = [ meson ninja pkg-config ];
```

File: pkgs/applications/video/obs-studio/obs-gstreamer.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, gstreamer
+, gst-plugins-base
+, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/obs-gstreamer.nix:6]:
```suggestion
, pkg-config
```

File: pkgs/applications/video/obs-studio/obs-gstreamer.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    sha256 = "1szfx5p2lb953blzw7prfd0nngjlwv2wn7jmnwvlssc9ci6jl6s8";
+  };
+
+  nativeBuildInputs = [ meson ninja pkgconfig ];
+  buildInputs = [ obs-studio gstreamer gst-plugins-base ];
+
+  meta = with lib; {
+    description = "GStreamer OBS Studio plugin";
+    homepage = "https://github.com/fzwoch/obs-gstreamer";
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/obs-gstreamer.nix:29]: Plus or Only?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by jansol** [comment]
This should benefit nicely from #125308

**Comment by yurkobb** [comment]
> This should benefit nicely from #125308

Oh, that's great news and will make using obs with plugins much easier! P.S. Regarding this PR, someone has [already packaged obs-gstreamer](https://github.com/NixOS/nixpkgs/blob/nixos-21.05/pkgs/applications/video/obs-studio/obs-gstreamer.nix#L38) in the meantime, so I think it can be closed.
